### PR TITLE
chore(authenticator): cherry-pick tests from main, add screen geometry tests

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Issue #, if available:*

*Description of changes:*
- Cherry-pick 9b2387c4e5a37629cdc45749068e45d4b6a4ddfd from main
- Update CognitoAuthStub for next branch
- Add tests for different screen geometries (screen size, pixel ratio)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_authenticator/test/ui/main_test.dart [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  defaultMaterial,
+  highContrast,
+  customSwatch,
+  custom,
+}
+
+/// Adds [lightTheme] and [darkTheme] getters, which cannot be properties since
+/// that are not constants.
+extension TestThemeX on TestTheme {
+  ThemeData get lightTheme {
```
Comment (dnys1) [inline packages/amplify_authenticator/test/ui/main_test.dart:72]: Can be directly on `TestTheme`

File: packages/amplify_authenticator/test/ui/main_test.dart [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+  final binding = TestWidgetsFlutterBinding.ensureInitialized();
+
+  setUp(() async {
+    // flutter's default test font does not accurately reflect font size,
+    // which leads to pixel over flows.
+    await loadAppFonts();
+
+    // default to mobile layout
+    binding.window.devicePixelRatioTestValue = ScreenGeometry.mobile.pixelRatio;
+    binding.window.physicalSizeTestValue = ScreenGeometry.mobile.size;
```
Comment (dnys1) [inline packages/amplify_authenticator/test/ui/main_test.dart:136]: Why are these needed if they get overridden below?

**Review by dnys1 — APPROVED** [review]
Nice work!

File: packages/amplify_authenticator/test/ui/main_test.dart (hunk 2 above)
Comment (Jordan-Nelson) [reply packages/amplify_authenticator/test/ui/main_test.dart]: Good catch. I had added it because I wasn't planning on adding ScreenGeometry as one of the dimensions for the theme tests, but changed my mind and added it. Updated.

**Review by Equartey — APPROVED** [review]
